cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java

```diff
@@ -180,145 +105,31 @@ private static DataprocClient getDataprocClient(DataprocConf conf, boolean requi
-                                                           + "Please create a subnet or use a different network.",
-                                                         network, networkHostProjectId));
-      }
-    }
-
-    subnet = chooseSubnet(network, subnets, subnet, conf.getRegion());
-
-    return new DataprocClient(new DataprocConf(conf, network, subnet), client, compute, networkInfo);
-  }
-
-  private static PeeringState getPeeringState(String systemProjectId, String systemNetwork, Network networkInfo) {
-    // note: vpc network is a global resource.
-    // https://cloud.google.com/compute/docs/regions-zones/global-regional-zonal-resources#globalresources
-    String systemNetworkPath = String.format("https://www.googleapis.com/compute/v1/projects/%s/global/networks/%s",
-                                             systemProjectId, systemNetwork);
-
-    LOG.trace(String.format("Self link for the system network is %s", systemNetworkPath));
-    List<NetworkPeering> peerings = networkInfo.getPeerings();
-    // if the customer does not has a peering established at all the peering list is null
-    if (peerings == null) {
-      return PeeringState.NONE;
-    }
-    for (NetworkPeering peering : peerings) {
-      if (!systemNetworkPath.equals(peering.getNetwork())) {
-        continue;
-      }
-      return peering.getState().equals("ACTIVE") ? PeeringState.ACTIVE : PeeringState.INACTIVE;
-    }
-    return PeeringState.NONE;
-  }
-
-  private static boolean subnetExists(List<String> subnets, String subnet) {
-    // subnets are of the form
-    // "https://www.googleapis.com/compute/v1/projects/<project>/regions/<region>/subnetworks/<name>"
-    // the provided subnet can be the full URI but is most often just the name
-    for (String networkSubnet : subnets) {
-      if (networkSubnet.equals(subnet) || networkSubnet.endsWith("subnetworks/" + subnet)) {
-        return true;
-      }
-    }
-    return false;
-  }
-
-  // subnets are identified as
-  // "https://www.googleapis.com/compute/v1/projects/<project>/regions/<region>/subnetworks/<name>"
-  // a subnet in the same region as the dataproc cluster must be chosen. If a subnet name is provided then the subnet
-  // will be choosen and the region will be picked on basis of the given zone. If a subnet name is not provided then
-  // any subnetwork in the region of the given zone will be picked.
-  private static String chooseSubnet(String network, List<String> subnets, @Nullable String subnet, String region) {
-    for (String currentSubnet : subnets) {
-      // if a subnet name is given then get the region of that subnet based on the zone
-      if (subnet != null && !currentSubnet.endsWith("subnetworks/" + subnet)) {
-        continue;
-      }
-      if (currentSubnet.contains(region + "/subnetworks")) {
-        return currentSubnet;
-      }
+      // Otherwise, make some Compute calls to pick a network
+      networkClient.init(null);
```

Comment:
over here, we still look up the network if null, even if SSH is not required.

I think I'll refactor this a bit, since we don't need to do this lookup for non-create operations.